### PR TITLE
GF 2.5 Updated System Status.

### PR DESCRIPTION
**Pull request by jestonihpi**
## Description
Added new Model for the system report.
Updated bootstrap file to inject the new Model.
Re design system report interface based on GF's 2.5 beta version.

## Testing instructions
Form -> System Status

## Screenshots 
image

## Checklist:
- [x] I've tested the code.
- [x] My code is easy to read, follow, and understand
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation / docblocks.

### CODE REVIEW

**Comment by jakejackson1** [comment]
@jestonihpi taking a quick look at this now. Here's the initial findings:

1. I noticed the Gravity PDF plugins (including extensions) aren't being included under the "Add-Ons" section in this PR yet. 
1. The WP Memory doesn't appear to be calculated correctly. I know for a fact my test site is configured with 512MB. 
1. The WP Memory doesn't have any error/warning message included if under 128MB.
1. `PHP PDF Working Directory` should be renamed `PDF Working Directory Path`. I think it best we split this up into `PDF Working Directory` and `PDF Working Directory URL`
1. `mPDF Temporary Protected` should be renamed "Temporary Folder Protected" and moved up one row.
1. `Temporary Folder Permissions` should have `777` changes to `Writable [tick]` or `Unwritable [cross]`, as per Gravity Forms. Also, move it up two rows so it's below the `Temporary Folder Location` row.
1. `Entry View Setting` should be renamed `PDF Entry List Action`
1. Let's remove the `.` at the end of the `Logged Out Timeout` value i.e `30 minutes` and not `30 minutes.`
1. For `Debug Mode Enabled` if it's turned off let's remove the [cross] and change it to `Off`. The cross makes it look like an error.
1. Type in `Security Settings` heading

**Comment by jakejackson1** [comment]
TODO:

1. Move Gravity PDF PLugin and Active Add-ons from "Active Plugins" box to "Add-Ons" box. 
1. Remove new Gravity PDF Add-ons box.

File: src/Model/Model_System_Report.php [hunk 1]
```diff
@@ -98,25 +99,25 @@ public function get_report_structure(): array {
 				'tables'       => [
 					[
 						'title'        => $language['table1'],
-						'title_export' => $language['table1'],
+						'title_export' => $title_export_prefix . $language['table1'],
```
Comment (jakejackson1) [inline src/Model/Model_System_Report.php:102]: The problem here is `$language['table1']` is still displaying translated text via `esc_html__( 'PHP', 'gravity-forms-pdf-extended' )`. You'll have to hardcode `PHP` instead. The same goes for the other `title_exports`.

File: src/Model/Model_System_Report.php [hunk 2]
```diff
@@ -133,16 +134,23 @@ public function get_report_structure(): array {
 	protected function get_report_items(): array {
 		$items = [];
 
+		$memory = $this->get_memory_limit();
+		$allow_url_fopen = $this->get_allow_url_fopen();
+		$temp_folder_protected = $this->check_temp_folder_permission();
+		$temp_folder_permission = $this->is_temporary_folder_writable();
```
Comment (jakejackson1) [inline src/Model/Model_System_Report.php:140]: You need to run this file through the linter. This would be a validation.

File: src/Model/Model_System_Report.php [hunk 3]
```diff
@@ -285,19 +296,30 @@ protected function prepare_report( $item ): array {
 			'label'        => $item['label'],
 			'label_export' => $item['label'],
 			'value'        => $item['value'],
-			'value_export' => $item['value'],
+			'value_export' => ( $item['value_export'] !== '' ) ?: $item['value'],
```
Comment (jakejackson1) [inline src/Model/Model_System_Report.php:299]: That's not how `?:` works.

File: src/Model/Model_System_Report.php [hunk 4]
```diff
@@ -285,19 +296,30 @@ protected function prepare_report( $item ): array {
...
-	protected function is_temporary_folder_writable(): string {
+	protected function is_temporary_folder_writable(): array {
 		$is_writable = wp_is_writable( $this->data->mpdf_tmp_location );
 
-		$output  = $is_writable ? __( 'Writable', 'gravityforms' ) : __( 'Not writable', 'gravityforms' );
-		$output .= $this->getController()->view->get_icon( $is_writable );
+		$string  = $is_writable ? __( 'Writable', 'gravityforms' ) : __( 'Not writable', 'gravityforms' );
+		$icon = $this->getController()->view->get_icon( $is_writable );
+
+		return ['value' => $string . $icon, 'value_export' => $string ];
```
Comment (jakejackson1) [inline src/Model/Model_System_Report.php:312]: `$string` is still translated when passed to `value_export`. Will need to hardcode the strings instead.

**Review by jakejackson1 — CHANGES_REQUESTED** [review]
Still some things that need to be corrected. I've marked them in the file.

File: tests/phpunit/unit-tests/View/Test_View_System_Report.php [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+	public function test_allow_url_fopen_success() {
+		$this->assertStringNotContainsString( 'is disabled', $this->view->get_allow_url_fopen( true ) );
+		$this->assertStringContainsString( 'is disabled', $this->view->get_allow_url_fopen( false ) );
+	}
+
+	public function test_get_icon_success() {
+		$this->assertStringContainsString( 'yes', $this->view->get_icon( true ) );
+		$this->assertStringContainsString( 'yes', $this->view->get_icon( 1 ) );
+		$this->assertStringContainsString( 'yes', $this->view->get_icon( 'string' ) );
```
Comment (jakejackson1) [inline tests/phpunit/unit-tests/View/Test_View_System_Report.php:56]: We'll need to add `declare( strict_types=1 );` to our View_System_Report, Model_System_Report, and Controller_System_Report files. These two tests should fail, as the `get_icon` method only accepts `bool` types.

File: tests/phpunit/unit-tests/View/Test_View_System_Report.php [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+	}
+
+	public function test_get_icon_failures() {
+		$this->assertStringContainsString( 'error', $this->view->get_icon( false ) );
+		$this->assertStringContainsString( 'error', $this->view->get_icon( 0 ) );
+
+	}
+
+	public function test_maybe_get_active_icon_empty() {
+		$this->assertEmpty( $this->view->maybe_get_active_icon( '' ) );
```
Comment (jakejackson1) [inline tests/phpunit/unit-tests/View/Test_View_System_Report.php:67]: You should also add a test that checks for the non-empty variant, too.

**Review by jakejackson1 — CHANGES_REQUESTED** [review]

File: tests/phpunit/unit-tests/Model/Test_Model_System_Report.php [hunk 7]
```diff
@@ -1,4 +1,5 @@
 <?php
+declare( strict_types=1 );
```
Comment (jakejackson1) [inline tests/phpunit/unit-tests/Model/Test_Model_System_Report.php:2]: Sorry, these declarations need to go in the non-test MVC classes. But you can leave them in the Test files too.

File: tests/phpunit/unit-tests/View/Test_View_System_Report.php [hunk 8]
```diff
@@ -52,18 +53,17 @@ public function test_allow_url_fopen_success() {
 
 	public function test_get_icon_success() {
 		$this->assertStringContainsString( 'yes', $this->view->get_icon( true ) );
-		$this->assertStringContainsString( 'yes', $this->view->get_icon( 1 ) );
-		$this->assertStringContainsString( 'yes', $this->view->get_icon( 'string' ) );
 
 	}
 
 	public function test_get_icon_failures() {
 		$this->assertStringContainsString( 'error', $this->view->get_icon( false ) );
```
Comment (jakejackson1) [inline tests/phpunit/unit-tests/View/Test_View_System_Report.php:60]: Because it's so simple, you could merge this test with the one above it.
